### ISSUE TITLE
build failure with unix-2.8

### ISSUE THREAD

**Issue by phadej** [TomMD/entropy#72]
```
Building library for entropy-0.4.1.7..
[1 of 2] Compiling System.EntropyNix ( System/EntropyNix.hs, /codetmp/e/entropy-0.4.1.7/.dist-newstyle-trustee/10be456790bc3b632af51d83d48f6261b1582f7afda9738a904fcdaae19022f73d8cecfb92dc8424569399c5837879f7b2e904384214c4b271bebb1b81838fcb/build/x86_64-linux/ghc-9.2.2/entropy-0.4.1.7/noopt/build/System/EntropyNix.o, /codetmp/e/entropy-0.4.1.7/.dist-newstyle-trustee/10be456790bc3b632af51d83d48f6261b1582f7afda9738a904fcdaae19022f73d8cecfb92dc8424569399c5837879f7b2e904384214c4b271bebb1b81838fcb/build/x86_64-linux/ghc-9.2.2/entropy-0.4.1.7/noopt/build/System/EntropyNix.dyn_o )


System/EntropyNix.hs:78:3: error:
    • Couldn't match expected type: unix-2.8.0.0:System.Posix.IO.Common.OpenFileFlags
                                    -> IO Fd
                  with actual type: IO Fd
    • The function ‘openFd’ is applied to four value arguments,
        but its type ‘FilePath
                      -> OpenMode
                      -> unix-2.8.0.0:System.Posix.IO.Common.OpenFileFlags
                      -> IO Fd’
        has only three
      In a stmt of a 'do' block:
        openFd source ReadOnly Nothing defaultFileFlags
      In the expression:
        do evaluate ensurePoolInitialized
           openFd source ReadOnly Nothing defaultFileFlags
   |
78 |   openFd source ReadOnly Nothing defaultFileFlags
   |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

System/EntropyNix.hs:78:26: error:
    • Couldn't match expected type ‘unix-2.8.0.0:System.Posix.IO.Common.OpenFileFlags’
                  with actual type ‘Maybe a0’
    • In the third argument of ‘openFd’, namely ‘Nothing’
      In a stmt of a 'do' block:
        openFd source ReadOnly Nothing defaultFileFlags
      In the expression:
        do evaluate ensurePoolInitialized
           openFd source ReadOnly Nothing defaultFileFlags
   |
78 |   openFd source ReadOnly Nothing defaultFileFlags
   |                          ^^^^^^^

```

I'll make corresponding revisions to affected versions (0.4 series, some of 0.3 too)

**Comment by athas** [comment]
Fixed by #69.

**Comment by TomMD** [comment]
thanks everyone.

See release 0.4.1.10
